survey/src/test/java/com/quickbirdstudios/test/TaskNavigatorTest.kt

```diff
@@ -0,0 +1,53 @@
+package com.quickbirdstudios.test
+
+import com.quickbirdstudios.surveykit.AnswerFormat
+import com.quickbirdstudios.surveykit.NavigableOrderedTask
+import com.quickbirdstudios.surveykit.OrderedTask
+import com.quickbirdstudios.surveykit.backend.navigator.NavigableOrderedTaskNavigator
+import com.quickbirdstudios.surveykit.backend.navigator.OrderedTaskNavigator
+import com.quickbirdstudios.surveykit.backend.navigator.TaskNavigator
+import com.quickbirdstudios.surveykit.steps.QuestionStep
+import org.junit.Assert
+import org.junit.Test
+
+class TaskNavigatorTest {
+    @Test
+    fun testHasNoPreviousStep() {
+        val firstQuestion = randomQuestionStep()
+        val secondQuestion = randomQuestionStep()
+        val thirdQuestion = randomQuestionStep()
+        val taskNavigator = createTaskImplementations(listOf(firstQuestion,secondQuestion,thirdQuestion))
+
+        for(nav in taskNavigator) {
+            nav.startStep(null)
+            Assert.assertFalse(nav.hasPreviousStep())
+        }
+    }
+
+    @Test
+    fun testHasPreviousStep() {
+        val firstQuestion = randomQuestionStep()
+        val secondQuestion = randomQuestionStep()
+        val thirdQuestion = randomQuestionStep()
+        val taskNavigator = createTaskImplementations(listOf(firstQuestion,secondQuestion,thirdQuestion))
+
+        for(nav in taskNavigator) {
+            nav.startStep(null)
+            nav.nextStep(thirdQuestion)
+            Assert.assertTrue(nav.hasPreviousStep())
+        }
+    }
+
+    private fun createTaskImplementations(steps: List<QuestionStep>) : List<TaskNavigator> {
+        return listOf(
+                OrderedTaskNavigator(OrderedTask(steps)),
+                NavigableOrderedTaskNavigator(NavigableOrderedTask(steps))
```

Comment:
You have to run the `ktlint` formatting over your changes

E.g. here the correct indentation would be only 8 spaces, not 12. And 2nd the closing `)` should be in a new line

Reply:
I have run the ktlint formatter on the code. The formatting should now be fine.